src/AppBundle/Issues/IssueListener.php

```diff
@@ -58,14 +58,39 @@ public function handleCommentAddedEvent($issueNumber, $comment)
      * Adds a "Needs Review" label to new PRs.
      *
      * @param int $prNumber The number of the PR
+     * @param int $prTitle  The title of the PR
+     * @param int $prBody   The full text description of the PR
      *
      * @return string The new status
      */
-    public function handlePullRequestCreatedEvent($prNumber)
+    public function handlePullRequestCreatedEvent($prNumber, $prTitle, $prBody)
     {
+        $prLabels = array();
+
+        // new PRs always require review
         $newStatus = Status::NEEDS_REVIEW;
+        $prLabels[] = $newStatus;
+
+        // the PR title usually indicates the affected component
+        if (preg_match('/^\[(?P<component>.*)\] .*/$', $prTitle, $matches)) {
```

Comment:
You are right. That regular expression was awful. I've updated everything. Thanks.